Common/ModelInfo.swift

```diff
@@ -0,0 +1,295 @@
+//
+//  ModelInfo.swift
+//  Diffusion
+//
+//  Created by Pedro Cuenca on 29/12/22.
+//  See LICENSE at https://github.com/huggingface/swift-coreml-diffusers/LICENSE
+//
+
+import CoreML
+
+enum AttentionVariant: String {
+    case original
+    case splitEinsum
+    case splitEinsumV2
+}
+
+extension AttentionVariant {
+    var defaultComputeUnits: MLComputeUnits { self == .original ? .cpuAndGPU : .cpuAndNeuralEngine }
+}
+
+/// Track if the model is downloaded and on the filesystem, ready for use.
+enum ModelReadinessState {
+    case unknown
+    case downloading
+    case downloaded
+    case uncompressing
+    case ready
+    case failed
+}
+
+class ModelReadiness: ObservableObject {
+    let modelInfo: ModelInfo
+    @Published var state: ModelReadinessState
+    
+    init(modelInfo: ModelInfo, state: ModelReadinessState) {
+        self.modelInfo = modelInfo
+        self.state = state
+    }
+}
+
+struct ModelInfo: Hashable, Identifiable {
```

Comment:
Why do we need `Hashable` and `Identifiable`?

Reply:
In the current code the Picker for the selected model uses the model's version, a String. SwiftUI List() does not like a String or something that can be a nil value as a selection type and can result in unknown behaviour. Along the way I was using the ModelInfo itself as the selected object, which required Hashable and Identifiable to be part of a list.
 In the latest commit of this PR the selection is actually an integer representing the index of the filtered array of known model info so I think that this can be removed.